File: src/main/java/itmo/is/project/dto/user/UpdateSpaceshipRequest.java
package itmo.is.project.dto.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import itmo.is.project.model.user.Spaceship;

public record UpdateSpaceshipRequest(

        @Schema(example = "S")
        @JsonProperty(value = "size", required = true)
        Spaceship.Size size
) {
}

File: src/main/kotlin/itmo/isproject/dto/user/spaceship/UpdateSpaceshipRequest.kt
package itmo.isproject.dto.user.spaceship

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import itmo.isproject.model.user.Spaceship

data class UpdateSpaceshipRequest(

    @Schema(example = "S")
    @JsonProperty(value = "size", required = true)
    val size: Spaceship.Size
)
